.github/workflows/docs-content-link-rewrites-e2e.yml

```
@@ -0,0 +1,71 @@
+name: docs-content-link-rewrites-e2e
+
+on:
+  workflow_call:
+    inputs:
+      repo-owner:
+        required: true
+        type: string
+        description: 'example: "hashicorp"'
+      repo-name:
+        required: true
+        type: string
+        description: 'example: "waypoint"'
+      commit-sha:
+        required: true
+        type: string
+        description: 'example: "github.sha"'
+      main-branch-preview-url:
+        required: true
+        type: string
+        description: 'example: "https://waypoint-git-main-hashicorp.vercel.app/"'
+      pr-branch-preview-url:
+        required: true
+        type: string
+        description: 'example: "https://waypoint-git-docs-ambmigrate-link-formats-hashicorp.vercel.app/"'
+
+jobs:
+  docs-content-link-rewrites-e2e:
+    runs-on: ubuntu-latest
+    env:
+      OWNER: ${{ inputs.repo-owner }}
+      REPO: ${{ inputs.repo-name }}
+      COMMIT_SHA: ${{ inputs.commit-sha }}
+      MAIN_BRANCH_PREVIEW_URL: ${{ inputs.main-branch-preview-url }}
+      PR_BRANCH_PREVIEW_URL: ${{ inputs.pr-branch-preview-url }}
+      ENABLE_DOCS_CONTENT_LINK_REWRITES_E2E: 'true'
+    steps:
+      - name: Determine if the workflow should run
+        id: should-run-for-branch
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const { data } = await github.rest.repos.listPullRequestsAssociatedWithCommit({
+              owner: process.env.OWNER,
+              repo: process.env.REPO,
+              commit_sha: process.env.COMMIT_SHA
+            })
+            return data && data[0].head.ref === "docs/amb.migrate-link-formats"
+
+      # @TODO remove `ref` setting before merging
+      - name: Check out dev-portal repo
+        uses: actions/checkout@v3
+        if: ${{ steps.should-run-for-branch.outputs.result == 'true' }}
+        with:
+          ref: amb.link-rewrites-e2e-tests
```

Will apply this suggestion after approval, and before merging

```suggestion
      - name: Check out dev-portal repo
        uses: actions/checkout@v3
        if: ${{ steps.should-run-for-branch.outputs.result == 'true' }}
```